intake/catalog/base.py

```diff
@@ -73,7 +75,11 @@ def __init__(self, *args, **kwargs):
             args = args[0]
         self.args = args
         self.updated = time.time()
-        self._entries = {}
+        # Allow the subclass to have set self._entries to some other dict-like
+        # object *before* this is run so that the last thing to happen is
+        # force_reload().
+        if getattr(self, '_entries', None) is None:
```

Comment:
This is necessary because of the way `__getitem__` and `__getattr__` mix. See for example:

```
In [49]: type(catalog)                                                                                                 
Out[49]: intake.catalog.base.RemoteCatalog

In [50]: hasattr(catalog, '_doesnotexit')                                                                              
Out[50]: True
```

I would definitely prefer to change this, but that's out of scope for this PR.

Reply:
This unexpected behavior is still an issue, but I did settle on a more elegant way to let subclasses customize `self._entries` in 9de555f7086a98f8ca3830ee82f07cdf0981f95a.